scripts/cybershake/auto_submit.py

```diff
@@ -19,78 +25,165 @@
 from scripts.submit_sim_imcalc import submit_im_calc_slurm, SlBodyOptConsts
 from shared_workflow import shared
 
-default_n_runs = {const.HPC.maui.value: 12, const.HPC.mahuika.value: 12}
-default_1d_mod = "/nesi/transit/nesi00213/VelocityModel/Mod-1D/Cant1D_v2-midQ_leer.1d"
-
-job_run_machine = {
-    const.ProcessType.EMOD3D.value: const.HPC.maui.value,
-    const.ProcessType.merge_ts.value: const.HPC.mahuika.value,
-    const.ProcessType.winbin_aio.value: const.HPC.maui.value,
-    const.ProcessType.HF.value: const.HPC.maui.value,
-    const.ProcessType.BB.value: const.HPC.maui.value,
-    const.ProcessType.IM_calculation.value: const.HPC.maui.value,
-    const.ProcessType.clean_up.value: const.HPC.mahuika.value,
+DEFAULT_N_MAX_RETRIES = 2
+DEFAULT_N_RUNS = {const.HPC.maui: 12, const.HPC.mahuika: 12}
+DEFAULT_1D_MOD = "/nesi/transit/nesi00213/VelocityModel/Mod-1D/Cant1D_v2-midQ_leer.1d"
+
+JOB_RUN_MACHINE = {
+    const.ProcessType.EMOD3D: const.HPC.maui,
+    const.ProcessType.merge_ts: const.HPC.mahuika,
+    const.ProcessType.winbin_aio: const.HPC.maui,
+    const.ProcessType.HF: const.HPC.maui,
+    const.ProcessType.BB: const.HPC.maui,
+    const.ProcessType.IM_calculation: const.HPC.maui,
+    const.ProcessType.clean_up: const.HPC.mahuika,
 }
 
+SLURM_TO_STATUS_DICT = {"R": 3, "PD": 2, "CG": 3}
+
+
+def get_queued_tasks(user=None, machine=const.HPC.maui):
+    if user is not None:
+        cmd = "squeue -A nesi00213 -o '%A %t' -h -M {} -u {}".format(
+            machine.value, user
+        )
+    else:
+        cmd = "squeue -A nesi00213 -o '%A %t' -h -M {}".format(machine.value)
+
+    process = Popen(shlex.split(cmd), stdout=PIPE, encoding="utf-8")
+    (output, err) = process.communicate()
+    process.wait()
+
+    output_list = list(filter(None, output.split("\n")[1:]))
+    return output_list
+
+
+def check_queue(queue_folder: str, run_name: str, proc_type: int):
```

Comment:
would probably be less confusing if we change the name a little bit, since now we have two 'queue': slurm queue and a PIPE queue for updaing management DB